src/legacy/core_plugins/data/server/lib/default_search_strategy.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Server, Request } from 'hapi';
+import { SearchOptions } from '../../common';
+import { registerSearchStrategy } from './search_strategies';
+import { getEsSearchConfig } from './get_es_search_config';
+
+export function registerDefaultSearchStrategy(server: Server) {
+  registerSearchStrategy('default', async function defaultSearchStrategy(
```

Comment:
nit: i would prefer exporting searchStrategy function here and doing registration `init`